src/app/api.py

```diff
@@ -61,10 +62,34 @@ async def request_summary(request: Request, call_next):
     return response
 
 
+@app.middleware("http")
+async def sentry_exception(request: Request, call_next):
+    """Middleware to report errors to Sentry"""
+    try:
+        response = await call_next(request)
+        return response
+    except Exception as exc:
+        with sentry_sdk.push_scope() as scope:
+            scope.set_context(
+                "request",
+                {
+                    "method": request.method,
+                    "path": request.url.path,
+                    "querystring": dict(request.query_params),
+                    "body": await request.body(),
+                },
+            )
+            sentry_sdk.capture_exception(exc)
+        raise exc
+
+
 @app.on_event("startup")
 def startup_event():
     """On app startup perform these setup operations"""
     configure_logging()
+    sentry_sdk.init(  # pylint: disable=abstract-class-instantiated  # noqa: E0110
+        dsn=settings.sentry_dsn
+    )
```

Comment:
[This comment](https://github.com/mozilla-it/ctms-api/blob/527d21d45922b3361325319487ab07254739a4eb/ctms/exception_capture.py#L11-L20) from the CTMS/FastAPI interactions with Sentry is a bit confusing to me. We may need this sentry.init to happen elsewhere--if I'm reading that comment correctly.

We can try this as is and make modifications if needed

Reply:
This startup callback was something you suggested [here](https://github.com/mozilla/jira-bugzilla-integration/pull/6#discussion_r813419676), but maybe it's simpler to do it during the app instantiation.

Reply:
Yeah, I think this is still the right location to startup the logging; but in CTMS--that comment linked above makes me wonder how threadsafe it is to do the actual `init` in this callback for sentry. 

In CTMS, we just add it to the app like this:
https://github.com/mozilla-it/ctms-api/blob/527d21d45922b3361325319487ab07254739a4eb/ctms/app.py#L107-L110

```    
init_sentry()
app.add_middleware(SentryAsgiMiddleware) ```

Reply:
In CTMS we only migrated Sentry out of the startup callback:

https://github.com/mozilla-it/ctms-api/blob/527d21d45922b3361325319487ab07254739a4eb/ctms/app.py#L113-L120

Reply:
I'm ok if we choose to resolve this comment; we can address whether we want to use the 'startup' callback in a later PR. 

(I'm still of the mind that Sentry is the outlier with this issue of using the startup callback.)